Sources/Brave/Frontend/UserContent/UserScripts/Scripts_Dynamic/Scripts/Paged/WalletSolanaProviderScript.js

```diff
@@ -60,27 +60,85 @@ window.__firefox__.execute(function($, $Object, $Function, $Array) {
           )
       }))
     })
-    /* <solanaWeb3.Transaction> ->
-      {transaction: <solanaWeb3.Transaction>,
-       serializedMessage: <base58 encoded string>,
-       signatures: [{publicKey: <base58 encoded string>, signature: <Buffer>}]} */
+    /*
+     solanaWeb3.Transaction | solanaWeb3.VersionedTransaction
+     ->
+     [UInt8]
+     */
+    let serializedMessageFromTx = $(function(transaction) {
+      if (transaction.message) { // VersionedTransaction (v0)
+        const serializeMessageBuffer = transaction.message.serialize();
+        const serializedMessage = [...serializeMessageBuffer]; // Buffer to Array
+        return serializedMessage;
+      } else { // Transaction (legacy)
+        const serializeMessageBuffer = transaction.serializeMessage();
+        const serializedMessage = [...serializeMessageBuffer]; // Buffer to Array
+        return serializedMessage;
```

Comment:
```suggestion
let serializeMessageBuffer = transaction.message ? transaction.message.serialize() : transaction.serializeMessage();
 const serializedMessage = [...serializeMessageBuffer]; // Buffer to Array
        return serializedMessage;
```

slight nit for improved readability

Reply:
Thanks, updated in [3d25fa4](https://github.com/brave/brave-ios/pull/7369/commits/3d25fa4e9b77f465e866d966b437927a2adad6b9)